### PR TITLE
citra_qt: enable per-monitor DPI scaling

### PR DESCRIPTION
**Pull request by BynariStar**
This fixes the scaling issues that occur when you have multiple monitors at different DPI and resolutions. There is still the issue with popup windows resizing themselves improperly when moved across screens.

Reference: http://doc.qt.io/qt-5/highdpi.html#high-dpi-support-in-qt

I would appreciate others testing this, just to make sure this doesn't have unintended side effects with other screen configurations.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/3907)

### CODE REVIEW

**Comment by wwylele** [comment]
Note: this bumps the Qt version requirement to 5.6, which is higher than the default one shipped with Ubuntu 16.04 (and probably some other linux distro). If we agree to add this now, please remember to revise [this guide](https://github.com/citra-emu/citra/wiki/Building-for-Ubuntu-16.04) or completely drop the support for it.

**Comment by wwylele** [comment]
@chris062689 That's only the case for the pre-build binary. We have never dropped the compilation support for older ubuntu and other linux, as long as it has Qt with sufficient version and other dependencies. I don't think only supporting ubuntu 18.04 is enough for you to say "we support linux"

File: src/citra_qt/main.cpp [hunk 1]
```diff
@@ -1447,6 +1447,7 @@ int main(int argc, char* argv[]) {
     QCoreApplication::setApplicationName("Citra");
 
     QApplication::setAttribute(Qt::AA_X11InitThreads);
+    QApplication::setAttribute(Qt::AA_EnableHighDpiScaling);
```
Comment (zhaowenlan1779) [inline src/citra_qt/main.cpp]: You can probably wrap this in:
```cpp
#if QT_VERSION >= QT_VERSION_CHECK(5, 6, 0)
   QApplication::setAttribute(Qt::AA_EnableHighDpiScaling);
#endif
```

**Comment by wwylele** [comment]
User reports that game icons are enlarged after this:

**Comment by BynariStar** [comment]
Did they provide any other information, like screen resolution(s), platform, and what it looks like on nightly?

**Comment by Robz8** [comment]
I do not have this issue.
My screen resolution is 1360x768, my platform is Windows 10, and looks the same on both canary and nightly.

**Comment by BynariStar** [comment]
Just to follow up here, the issue the user had with enlarged icons was just due to their Windows scaling being set too high.

**Comment by BynariStar** [comment]
If by 150% DPI you meaning Windows scaling, then that would be the same cause as what @wwylele reported. As that's an issue with the user's specific settings, I don't intend on adding an option to disable hidpi support within Citra.

If it really bothers them and they can't find a solution (i.e. lowering scaling, disabling Windows' scaling overrides) they can add `QT_AUTO_SCREEN_SCALE_FACTOR=0` to their environment variables.

Can you confirm the status bar issue is caused by this PR specifically?

**Comment by wwylele** [comment]
@BynariStar I think the 150% scale is the default setting Windows gives to high DPI screens, and we shouldn't blame users for not changing it / not adding an environment variable.

**Comment by matiffeder** [comment]
I'm the one who report to zhaowenlan1779, the status bar is not hidden, is covered, because I can't make the height of Citra Window shorter, please watch the video that zhaowenlan1779 uploaded to more information. I hope it will be able to disable this function, I think there are also many people use 150% scale, because laptop always have a small screen.

**Comment by BynariStar** [comment]
@matiffeder I'm not really sure what's causing that issue. I took a quick look at Dolphin's implementation, and they're using the same attribute. Perhaps someone more acquainted with Qt will be able to chime in.

**Comment by wwylele** [comment]
Pulling this from canary for now as I received more similar issue reports. Will add it back on next change.

**Comment by BynariStar** [comment]
@wwylele please detail the reports here

**Comment by jroweboy** [comment]
I'd love for someone to take some time to play around with all sorts of strange DPI combinations and see if they can get this to work right :) If you are still planning to work on this, feel free to push some commits and leave a comment here and we can re open this 👍

**Pull request closed without merging**